### PR TITLE
HIVE-20316: Skip external table file listing for create table event.

### PR DESCRIPTION
**Pull request by sankarh**

### CODE REVIEW

File: hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java [hunk 1]
```diff
@@ -155,9 +156,11 @@ public void onConfigChange(ConfigChangeEvent tableEvent) throws MetaException {
   @Override
   public void onCreateTable(CreateTableEvent tableEvent) throws MetaException {
     Table t = tableEvent.getTable();
+    FileIterator fileIter = t.getTableType().equals(TableType.EXTERNAL_TABLE.toString())
+                              ? null : new FileIterator(t.getSd().getLocation());
```
Comment (maheshk114) [inline hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java]: if external table dump is enabled then we should allow file listing or do something to get the file list during dump

File: hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java [hunk 2]
```diff
@@ -155,9 +156,11 @@ public void onConfigChange(ConfigChangeEvent tableEvent) throws MetaException {
...
     NotificationEvent event =
-        new NotificationEvent(0, now(), EventType.CREATE_TABLE.toString(), msgFactory
-            .buildCreateTableMessage(t, new FileIterator(t.getSd().getLocation())).toString());
+        new NotificationEvent(0, now(), EventType.CREATE_TABLE.toString(),
+                msgFactory.buildCreateTableMessage(t, fileIter).toString());
```
Comment (maheshk114) [inline hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java]: if we are passing table object to buildMessage method, then why need to check here for external table ?

File: itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationListener.java [hunk 3]
```diff
@@ -428,7 +428,7 @@ public void createTable() throws Exception {
     // a failed event should not create a new notification
     table =
         new Table(tblName2, defaultDbName, tblOwner, startTime, startTime, 0, sd, null,
-            emptyParameters, null, null, null);
+            emptyParameters, null, null, TableType.MANAGED_TABLE.toString());
```
Comment (maheshk114) [inline itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationListener.java]: why we need it ? does not it by default managed table ?

File: hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java (hunk 1 above)
Comment (maheshk114) [reply hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java]: i think its looks odd that create table will not dump the files but insert will ..i think we should stop both or allow both

Comment (sankarh) [reply hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java]: ok, will remove file listing even for inserts.

**Pull request closed without merging**